pkg/generate/code/set_resource.go

```diff
@@ -219,6 +206,53 @@ func SetResource(
 		}
 
 		targetMemberShapeRef = f.ShapeRef
+
+		// We may have some special instructions for how to handle setting the
+		// field value...
+		setCfg := f.GetSetterConfig(opType)
+
+		if setCfg != nil && setCfg.Ignore {
+			continue
+		}
+
+		sourceMemberShapeRef := outputShape.MemberRefs[memberName]
+		if sourceMemberShapeRef.Shape == nil {
+			// We may have some instructions to specially handle this field by
+			// accessing a different Output shape member.
+			//
+			// As an example, let's say we're dealing with a resource called
+			// Bucket that has a "Name" field. In the Output shape of the
+			// ReadOne resource manager method, the field name is "Bucket",
+			// though. There may be a generator.yaml file that looks like this:
+			//
+			// resources:
+			//   Bucket:
+			//     fields:
+			//       Name:
+			//         set:
+			//          - method: ReadOne
+			//            from: Bucket
+			//
+			// And this would indicate to the code generator that the Spec.Name
+			// field should be set from the value of the Output shape's Bucket
+			// field.
+			var found bool
+			if setCfg != nil && setCfg.From != nil {
+				sourceMemberShapeRef, found = outputShape.MemberRefs[*setCfg.From]
```

Comment:
Do we want to supported nested field paths here, as well?

Reply:
+1, I'm fairly certain I will need nested field paths support for DhcpOptions. To summarize, DhcpOptions Spec has a _slice of_  `*NewDHCPConfiguration`:

```
type NewDHCPConfiguration struct {
	Key *string `json:"key,omitempty"`
	Values []*string `json:"values,omitempty"`
}
```

and the CreateDhcpOptions Result output shape returns a _slice of_ `*DHCPConfiguration`:

```
type AttributeValue struct {
	Value *string `json:"value,omitempty"`
}

type DHCPConfiguration struct {
	Key *string `json:"key,omitempty"`
	Values []*AttributeValue `json:"values,omitempty"`
}
```

I need entries in `NewDHCPConfiguration.Values` to be populated with data from  `DHCPConfiguration.Values.Value`